ee/clickhouse/queries/funnels/funnel_trends.py

```diff
@@ -92,11 +92,18 @@ def get_query(self) -> str:
 
         reached_from_step_count_condition, reached_to_step_count_condition, _ = self.get_steps_reached_conditions()
         interval_method = get_trunc_func_ch(self._filter.interval)
+
+        if self._filter.date_from is None:
```

Comment:
Q: Should the filter object handle this business logic?

Reply:
There is some handling happening there: https://github.com/PostHog/posthog/blob/master/posthog/models/filters/mixins/common.py#L219

but this just looks weird to me now. (Too many places trying to handle this). Agree with making it one place, although don't think it's possible in the filter object, since no team attribute there.

Looks like lots of discrepancies, but this one also handles "all" in the mixin: https://github.com/PostHog/posthog/blob/master/posthog/models/filters/mixins/stickiness.py#L33

Reply:
there should be some shared way to handle this. We've ended up in a situation where the time bucketing is handled slightly differently in different places. for trends, we go from date_to backwards while in this case we're going from date_from forwards

I think this refactor belongs in a separate PR though